_inc/lib/admin-pages/class.jetpack-admin-page.php

```diff
@@ -41,6 +46,10 @@ function __construct() {
 		self::$block_page_rendering_for_idc = (
 			Jetpack::validate_sync_error_idc_option() && ! Jetpack_Options::get_option( 'safe_mode_confirmed' )
 		);
+		if( $_GET[ 'page' ] == 'jetpack' ) {
+			add_filter( 'nocache_headers', 'add_no_store_header' );
```

Comment:
Since other plugins and themes can also modify this, we might want to add it with a larger `$priority`, so it would be called a bit later in the filter application call queue.